src/rust/bitbox02-rust/src/hww/api/ethereum/sign.rs

```diff
@@ -70,16 +133,59 @@ fn parse_erc20(request: &pb::EthSignRequest) -> Option<([u8; 20], BigUint)> {
 }
 
 // fee: gas limit * gas price:
-fn parse_fee<'a>(request: &pb::EthSignRequest, params: &'a Params) -> Amount<'a> {
-    let gas_price = BigUint::from_bytes_be(&request.gas_price);
-    let gas_limit = BigUint::from_bytes_be(&request.gas_limit);
-    Amount {
-        unit: params.unit,
-        decimals: WEI_DECIMALS,
-        value: gas_price.mul(gas_limit),
+fn parse_fee<'a>(request: &Transaction<'a>, params: &'a Params) -> Amount<'a> {
```

Comment:
Use `Transaction<'_>` not `'a` - the  resulting Amount lifetime only depends on the lifetime of `params`.